### PR TITLE
Port coding-system-aliases and check-coding-system

### LINKED ISSUE #1173: Port coding-system-aliases
```c
DEFUN ("coding-system-aliases", Fcoding_system_aliases, Scoding_system_aliases,
       1, 1, 0,
       doc: /* Return the list of aliases of CODING-SYSTEM.  */)
  (Lisp_Object coding_system)
{
  Lisp_Object spec;

  if (NILP (coding_system))
    coding_system = Qno_conversion;
  CHECK_CODING_SYSTEM_GET_SPEC (coding_system, spec);
  return AREF (spec, 1);
}
```

### PR DESCRIPTION
**Pull request by daichimukai**
Closes #1173.

### CODE REVIEW

**Comment by daichimukai** [comment]
The build failed, something still wrong with LispMarkerRef or releated ones?
```
thread '<unnamed>' panicked at 'assertion failed: charpos <= bytepos', src/marker.rs:351:13
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:59
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:476
   5: std::panicking::begin_panic
   6: remacs::marker::set_marker_internal
   7: Fset_marker
   8: Fmatch_data
             at /home/travis/build/Wilfred/remacs/src/search.c:2653
...
```

File: rust_src/src/coding.rs [hunk 1]
```diff
@@ -19,6 +37,21 @@ pub fn coding_system_id(object: LispObject) -> isize {
...
+    if spec.is_nil() {
+        check_coding_system_lisp(x);
+        spec = coding_system_spec(x);
+        if spec.is_nil() {
+            wrong_type!(Qcoding_system_p, x);
+        }
+    }
+    spec
+}
+
```
Comment (shaleh) [inline rust_src/src/coding.rs]: An alternative implementation for the previous function would be:

```
fn check_coding_system_get_spec(x: LispObject) -> LispObject {
    match coding_system_spec(x) {
        Qnil => {
            check_coding_system_lisp(x);
            match coding_system_spec(x) {
                Qnil => wrong_type!(Qcoding_system_p, x),
                spec => spec,
            }
        }
        spec => spec,
    }
}
```

Comment (shaleh) [reply rust_src/src/coding.rs]: I am not certain which I prefer.

File: rust_src/src/coding.rs [hunk 2]
```diff
@@ -29,4 +62,32 @@ pub fn coding_system_p(object: LispObject) -> bool {
...
+#[lisp_fn]
+pub fn coding_system_aliases(coding_system: LispObject) -> LispObject {
+    let mut coding_system = coding_system;
+    if coding_system.is_nil() {
+        coding_system = Qno_conversion;
+    }
+    let spec = check_coding_system_get_spec(coding_system);
+    aref(spec, 1)
+}
+
```
Comment (shaleh) [inline rust_src/src/coding.rs]: The above could also be written:

```
#[lisp_fn]
pub fn coding_system_aliases(coding_system: LispObject) -> LispObject {
    let coding_system = match coding_system {
        Qnil => Qno_conversion,
        val => val,
    };
    let spec = check_coding_system_get_spec(coding_system);
    aref(spec, 1)
}
```

File: rust_src/src/coding.rs [hunk 3]
```diff
@@ -3,12 +3,30 @@
...
-    remacs_sys::{Fget, Qcoding_system_define_form, Vcoding_system_hash_table},
+    lists::{get, put},
+    remacs_sys::{
+        safe_eval, Fget, Qcoding_system_define_form, Qcoding_system_error, Qcoding_system_p, Qnil,
+        Qno_conversion, Vcoding_system_hash_table,
+    },
 };
 
+/// Return the spec vector of CODING_SYSTEM_SYMBOL.
+/// Same as the CODING_SYSTME_SPEC C macro.
```
Comment (nickdrozd) [inline rust_src/src/coding.rs]: Typo: SYSTME -> SYSTEM

**Comment by Wilfred** [comment]
@daichimukai I've sent you an invite to be a collaborator! This lets you approve other people's PRs if you want to :)

**Comment by daichimukai** [comment]
@Wilfred Thanks for the invitation! I'm glad to be a collaborator. I would like to start by leaving some comments and questions on PRs.

**Pull request merged**